Make sure data examples achieve a single task

dstansby opened the pull request:
As part of the documentation reorganisation we will have the four types of documentation described at https://diataxis.fr/. One of these is "How-to guides", which are 
> recipes, directions that guide the reader through the steps to achieve a specific end.

We already have these guides to some extent in our example gallery.

This PR is the first part of work to make sure the files in our example gallery achieve **one** specific task. I don't think any of the delted content in this PR needs to be moved elsewhere as it's covered by other examples, but in PRs for the other example categories we may want to move some stuff to entries in a Tutorials section.

Fixes #5888

nabobalis commented on examples/acquiring_data/downloading_lascoC2.py:
```diff
@@ -1,41 +0,0 @@
-"""
```
👍 

I think back in the day we had this due to the metadata needing to be manually corrected.

nabobalis commented on examples/acquiring_data/goes_xrs_example.py:
```diff
@@ -69,37 +66,6 @@
 goes_15 = ts.TimeSeries(file_goes15)
 goes_15.peek()
 
-#############################################################
-# The resulting `~sunpy.timeseries.TimeSeries` can be filtered by GOES quality flags. For more information
-# refer to the `GOES Data Guide <https://satdat.ngdc.noaa.gov/sem/goes/data/science/xrs/GOES_13-15_XRS_Science-Quality_Data_Readme.pdf>`__.
-
-df = goes_15.to_dataframe()
-df = df[(df["xrsa_quality"] == 0) & (df["xrsb_quality"] == 0)]
-goes_15 = ts.TimeSeries(df, goes_15.meta, goes_15.units)
```
I wonder if this needs to stay but this becomes a tutorial or something? Or maybe text in the GOES timeseries class about the quality flags?

dstansby replied:
👍 I can move this to a temporary folder for future tutorial material so it's not lost

nabobalis commented on examples/map_transformations/downloading_hmi.py:
```diff
@@ -1,9 +1,10 @@
-Downloading and plotting an HMI magnetogram
-===========================================
+==============================================
+Rotating HMI maps so they're not 'upside-down'
```
```suggestion
Solar physicists hate it, One Simple Trick to rotate your upside-down HMI image
```
I think this needs to be more Buzzfeed style

dstansby replied:
> They opened a HMI FITS file and you won't believe what happened next

nabobalis commented on examples/acquiring_data/downloading_cutouts.py:
```diff
@@ -92,23 +91,4 @@
-for each_map in sequence:
-    each_map.plot_settings['norm'] = ImageNormalize(vmin=0, vmax=5e3, stretch=SqrtStretch())
-plt.figure()
-ani = sequence.plot()
```
We lose one of our only gallery animations tho and I am sad to see that go.

dstansby replied:
Other suggestions welcome - "Plotting an animation" seems like it's a single task that only needs one example though 😢  - we can fill up the future tutorials with them though!

nabobalis replied:
Yeah, it should go, we should check if we have an animation sequence example somewhere and not if, a new PR/issue to add one in future.

dstansby replied:
https://docs.sunpy.org/en/stable/generated/gallery/plotting/arrayanimatorwcs.html#sphx-glr-generated-gallery-plotting-arrayanimatorwcs-py already exsists 👍

dstansby commented on examples/acquiring_data/downloading_cutouts.py:
```diff
@@ -92,23 +91,4 @@
 
 files = Fido.fetch(query)
```
Is this hot take just to make the example pass? If so I'm 👍 (and I can rename the example to "searching for cutouts")

dstansby commented:
Sorry for the reviewer pain, I've updated my commit and force pushed. All your comments have been addressed - might be worth having a particular look at the cutout example, which I've modified quite a bit. Tested and working locally for me.

nabobalis commented:
I think a tracking issue for the tutorials folder should be opened once this is merged so we don't forget.

nabobalis commented:
Needs a second review but LGTM

Cadair commented on examples/acquiring_data/downloading_lascoC2.py:
```diff
@@ -1,41 +0,0 @@
-"""
-======================================
-Downloading and plotting LASCO C2 data
```
I get that this isn't doing anything particularly "interesting" but if someone wants to get a C2 image they are going to find this and follow it. What's the harm in keeping it?

Cadair commented on examples/acquiring_data/downloading_cutouts.py:
```diff
@@ -78,37 +59,16 @@
+print(query)
 
 #####################################################
-# Submit the export request and download the data.
-
-files = Fido.fetch(query)
-files.sort()
-
-#####################################################
-# Now that we've downloaded the files, we can create
-# a `~sunpy.map.MapSequence` from them.
-
-sequence = sunpy.map.Map(files, sequence=True)
```
Why not also show download and load into a Map? It's all part of a single "task" in my opinion, you aren't going to want to search for and download stuff without wanting to load them after?

Cadair commented on examples/acquiring_data/downloading_cutouts.py:
```diff
@@ -78,37 +59,16 @@
-
-sequence = sunpy.map.Map(files, sequence=True)
-
-#####################################################
-# Finally, we can construct an animation in time from
-# our stack of cutouts and interactively flip through
-# each image in our sequence. We first adjust the plot
-# settings on each image to ensure the colorbar is the
-# same at each time step.
-
-for each_map in sequence:
-    each_map.plot_settings['norm'] = ImageNormalize(vmin=0, vmax=5e3, stretch=SqrtStretch())
-plt.figure()
-ani = sequence.plot()
-
-plt.show()
```
The only reason I am sad about removing this is that this was a nice little gif on the gallery page, and much more eye catching than the now complete lack of a preview image.

wtbarnes replied:
FWIW, this code can now be greatly simplified to just set `norm=...` in `sequence.plot()`.

dstansby replied:
Is that a blocking request for this PR? I'd advocate for merge this as it's an improvement and this plotting code isn't changed, and we can sort of all the norms (if there are more?) in another PR.

wtbarnes replied:
Nope not blocking. I should've cleaned these up a long time ago.

Cadair reviewed (commented):
While I get the desire to be pure to the methodology, I do think we also need to consider the experience of someone scrolling through the gallery, if you make a plot after you search for / download data then you get a nice preview image where someone can say "hey that looks like what I am after" if we always just stop at `Fido.fetch` then you don't get that.

This is particularly relevant to the JSOC cutout example here.

dstansby commented:
> if you make a plot after you search for / download data then you get a nice preview image where someone can say "hey that looks like what I am after" if we always just stop at `Fido.fetch` then you don't get that.

👍 good point - I will go through these and put back in or add new simple plots wherever we download data.

dstansby commented:
No, I don't think we should either! This PR now doesn't remove any final images.

wtbarnes commented on examples/map_transformations/downloading_hmi.py:
```diff
@@ -1,9 +1,10 @@
-Downloading and plotting an HMI magnetogram
-===========================================
+==============================================
+Rotating HMI maps so they're not 'upside-down'
```
👍

nabobalis commented:
Doc failure on: 
"/home/runner/work/sunpy/sunpy/docs/guide/data_types/maps.rst:26: WARNING: undefined label: 'sphx_glr_generated_gallery_acquiring_data_2011_06_07_sampledata_overview.py'"

Cadair commented:
This should have had two approvals.

Cadair commented on examples/map_transformations/downloading_hmi.py:
```diff
@@ -1,9 +1,10 @@
 
-This example shows how to download a HMI magnetogram data with Fido and make a plot.
+This example shows how to rotate a HMI magnetogram, so when you plot it
+it appears with solar North puting up, and not upside down!
```
```suggestion
it appears with solar North pointing up.
```

Cadair commented on examples/acquiring_data/downloading_cutouts.py:
```diff
@@ -34,32 +34,16 @@
-cutouts.plot()
-
-plt.show()
+# Next, we will use this map to definte the top right and bottom
```
```suggestion
# Next, we will use the coordinate frame from this map to definte the top right and bottom
```

Cadair commented on examples/acquiring_data/downloading_cutouts.py:
```diff
@@ -34,32 +34,16 @@
-
-plt.show()
+# Next, we will use this map to definte the top right and bottom
+# left coordinates we want for the submap.
```
```suggestion
# left coordinates we want for the cutout request.
```

nabobalis commented:
> This should have had two approvals.

The original version for sure.

Cadair commented:
The original version of what?

nabobalis commented:
> The original version of what?

Of this PR

Cadair commented:
There was another version of this PR? I am very confused.

nabobalis commented:
It's ok.
